### ISSUE TITLE
UI elements can only be in one place

### ISSUE THREAD

**Issue by buep** [Praqma/pretested-integration-plugin#56]
We moved the plugin to be a git extension plugin, so this breaks backwards compatibility and we need to remove the old UI as the same configuration is now done as a git extension.

**Comment by buep** [comment]
Giving could have because what we have works, but clean-up and one consistent interface will be better.

Current work from @bicschneider doesn't break backwards compatibility but cleaning up UI will.

If we do not clean-up UI, I will suggest the following different delivery:

* a warning to the user in the Jenkins UI about next version will not be compatible
* a warning on next major version where we set a notifcation to the user when upgrading - I have seen plugin mark themselves not compatible when upgrading (somehow).

**Comment by buep** [comment]
@lakruzz - will you really want to clean up the UI this time?

Estimate to clean up is about 8 days.

as we also need to clean-up the git bridge so this is part of the git extension, while all the functionality we (wrongly?) placed in the git bridge need to go somewhere.

**Comment by bicschneider** [comment]
Currently it is done so that a "doubled" configuration will result in this failure with PiP Build Wrapper is being executed:
```
[PREINT]  Pretested Integration Plugin v2.4.2-SNAPSHOT (private-213fef94-Claus)
[PREINT]  ERROR - setUp() - UnsupportedConfigurationException - You have configurated Git Phlow Integration as Git Plugin Extension already.
It does make sense to have both. Please remove it in the Build Environment section
```
Error message could be refined

**Comment by groenborg** [comment]
I have removed the extra configuration block in Jenkins, by setting both to false. @MadsNielsen said that it should be enough to remove the bullet in Jenkins. It however might leave som legacy code behind! That needs investigating 

#### change
https://github.com/Praqma/pretested-integration-plugin/blob/56-ui-elements-can-only-be-in-one-place/src/main/java/org/jenkinsci/plugins/pretestedintegration/PretestedIntegrationBuildWrapper.java#L148-L154

**Comment by buep** [comment]
The line you reference above doesn't remove or disable the UI elements, but decide if our when the plugin is applicable based on the type of job.

We want the old UI not to show anywhere. So it needs to clean up some jelly files, and the bits and pieces that uses the input fields in the UI.

I'll ask @lakruzz one last time if he want the breaking change, or if Claus test on if the user configuration is done twice is good enough.
Should we do this breaking change independent from the cost? Hours spent, testing effort, documentation effort etc. ?

**Comment by lakruzz** [comment]
@buep 

I'm thinking like this:

* Having dual entries in the GUI does not make much sense - people might stop people taking us serious - I'm thinking about our brand!
* We no that _at some point in the future_ we will have to get rid of it. And we know that we will never be able to finde a customer, that will pay for us cleaning up after our own mess - so waiting doesn't help us the slightest bit.
* Breaking backwards compatibility might not be broken - if it's only a GUI element - we can claim, that moving from one position in the GUI to another is not about being compatible - it'a about lay-out and usability.

Is it _really_ going to take 8 days (56 hours!)  of work to clean up Jelly files?

What are our options - I feel like I'm a bit taken hostage here between "Do The Right Thing - It will cost you a fortune" and "Sell out on all your principles and release a hack"

Is there really no middleground?

**Comment by buep** [comment]
> Breaking backwards compatibility you say - but it might not be broken - if it's only a GUI element - we can claim, that moving from one position in the GUI to another is not about being compatible - it'a about lay-out and usability.
- We have done this before in another plugin, but it requires a two stop upgrade from the users side. First version will make sure old configuration is moved to the new one, the second upgrade will switch to the new one. Thierry did something similar in memory-map, so I think we could migrate the users automatically. If we really like - but I'm more in favor of just breaking backwards compatibility because most users these days doesn't make job in the UI so it is easy to update their job dsl script to the new way.

> Is it really going to take 8 days (56 hours!) of work to clean up Jelly files?
Estimate isn't mine, I think it is mostly because we have to change a large amount of tests. I don't know how that estimate came up.

> What are our options - I feel like I'm a bit taken hostage here between "Do The Right Thing - It will cost you a fortune" and "Sell out on all your principles and release a hack"

I just wanted your input on what you think is best from a user perspective... I was in doubt. I could see the benefit from the users side of having both and explain to the user they shouldn't use both and should migrate to the new one.

**Comment by buep** [comment]
Will be solved with #81 where we actually remove the code behind it... doesn't seem so much work as someone earlier suggested.
